firecrown/connector/numcosmo/numcosmo.py

```diff
@@ -0,0 +1,386 @@
+"""Support of Firecrown likelihood use in NumCosmo.
+
+The subpackages and modules in this package depend upon NumCosmo, and can not
+be used without an installation of NumCosmo.
+
+"""
+
+from typing import Dict, Union, List, Any, Optional
+import numpy as np
+import pyccl as ccl
+
+import gi
+
+gi.require_version("NumCosmo", "1.0")
+gi.require_version("NumCosmoMath", "1.0")
+
+# pylint: disable=wrong-import-position
+from gi.repository import NumCosmo as Nc  # noqa: E402
+from gi.repository import NumCosmoMath as Ncm  # noqa: E402
+
+from firecrown.likelihood.likelihood import load_likelihood, Likelihood  # noqa: E402
+from firecrown.likelihood.gauss_family.gaussian import ConstGaussian  # noqa: E402
+from firecrown.parameters import ParamsMap  # noqa: E402
+from firecrown.connector.mapping import Mapping, build_ccl_background_dict  # noqa: E402
+
+# pylint: enable=wrong-import-position
+
+
+class MappingNumCosmo(Mapping):
+    """Mapping support for NumCosmo."""
+
+    def __init__(
+        self,
+        *,
+        require_nonlinear_pk: bool = False,
+        p_ml: Optional[Nc.PowspecML] = None,
+        p_mnl: Optional[Nc.PowspecMNL] = None,
+        dist: Nc.Distance,
+    ):
+        super().__init__(require_nonlinear_pk=require_nonlinear_pk)
+        self.p_ml = p_ml
+        self.p_mnl = p_mnl
+        self.dist = dist
+
+    def get_params_names(self):
```

Comment:
We should remove these 4 functions and make them non-virtual in the base class. We should also deprecate them in the base class, preparing to remove them from the public interface in firecrown 2.0

Reply:
Done!